docs/topics/auth/username.txt

```diff
@@ -0,0 +1,130 @@
+=============================
+Username Validation in Django
+=============================
+
+.. module:: django.contrib.auth.username_validation
+
+Django offers pluggable username validation.You can configure multiple
+username validators at the same time. A few validators are included in 
+Django, but you can write your own as well.
+
+Each username validator must provide a help text to explain the requirements to
+the user, validate  a given username and return an error message if it does not 
+meet the requirements. Validators can also have optional settings to fine tune
+their behavior.
+
+Validation is controlled by the :setting:`AUTH_USERNAME_VALIDATORS` setting.
+The default for the setting is an ``UnicodeUsernameValidator`` which
+validates whether given username contains English letters, numbers and @/./+/-/_
+characters or not.
+
+By default, validators are used in the forms and in the 
+:djadmin:`createsuperuser` management commands.
+
+Enabling username validation
+----------------------------
+
+Username validation is configured in the
+:setting:`AUTH_USERNAME_VALIDATORS` settings::
+
+    AUTH_USERNAME_VALIDATORS = [
+        {
+            'NAME': 'django.contrib.auth.validators.UnicodeUsernameValidators',
+        },
+        {
+            'NAME': 'django.contrib.auth.validators.UsernameMinimumLengthValidator',
+            'OPTIONS':{
+                'min_length':3,
+            }
+        }
+    ]
+
+This example enables two included validators:
+
+* ``UnicodeUsernameValidator``, which checks whether username contains
+  anything other than the English letters, numbers and @/./+/-/_.
+* ``UsernameMinimumLengthValidator``, which checks whether the username meets a minimum
+  length. This validator is configured with a custom option it now requires
+  the minimum length to be nine characters, instead of the default eight.
```

Comment:
the minimum length to be three characters, instead of the default zero.

which sort of raises the question whether the default should be something other than zero!

Reply:
Recent changes look good! This one still says nine characters instead of the default eight when I think it should be: three characters instead of the default one.

Reply:
I have fixed this issue and committed code. Forgot to mention here.